libnavui-androidauto/src/main/java/com/mapbox/androidauto/notification/CarNotificationInterceptor.kt

```diff
@@ -0,0 +1,135 @@
+package com.mapbox.androidauto.notification
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import androidx.car.app.model.CarColor
+import androidx.car.app.notification.CarAppExtender
+import androidx.car.app.notification.CarPendingIntent
+import androidx.core.app.NotificationCompat
+import com.mapbox.androidauto.R
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.TimeFormat
+import com.mapbox.navigation.base.formatter.DistanceFormatter
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.formatter.MapboxDistanceFormatter
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.core.trip.session.NavigationSessionState
+import com.mapbox.navigation.core.trip.session.NavigationSessionStateObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+
+/**
+ * Register this observer using [MapboxNavigationApp.registerObserver]. As long as it is
+ * registered, the trip notification will be shown in the car.
+ */
```

Comment:
Would like to point people to the `setTripNotificationInterceptor` in case they're looking to make a notification interceptor that is very specific to their domain. For example, "Passenger canceled" is a difficult case for us to support here. 

```suggestion
 * registered, the trip notification will be shown in the car using
 * [MapboxNavigation.setTripNotificationInterceptor].
 */
```